addon-test-support/ember-qunit/test-isolation-validation.js

```diff
@@ -65,8 +63,20 @@ export function installTestNotIsolatedHook() {
   //    - it detatches the failure from the actual test that failed, making it
   //      more confusing to the end user.
   test.finish = function() {
+    let doFinish = () => finish.apply(this, arguments);
+
     detectIfTestNotIsolated(this);
 
-    return finish.apply(this, arguments);
+    if (isSettled()) {
+      return doFinish();
+    } else {
+      return waitUntil(isSettled, { timeout: 100 })
```

Comment:
I've since changed this to `Infinity`, as it's clear to me via testing that any arbitrary timeout we set here won't be adequate. If a user schedules something that takes over the arbitrary value, this test will timeout. We likely want to wait until it's properly settled before continuing.

Thoughts?

Reply:
I've changed this back to `100`, and fixed any issues that were resulting from the change to `waitUntil`. This should be good to go now.